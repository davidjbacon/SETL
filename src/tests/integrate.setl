const dx = 0.01;

f_ref := routine f;
assert f_ref = (routine) f;  -- the "cast"-like syntax
area := integrate(f_ref, 0, 1);
print(area);
rite := -cos 1 - -cos 0;
print(rite);
print(area - rite);

proc f(x);
  return sin x;
end f;

proc integrate(g, lo, hi);  -- not robust or accurate, but who cares
  n := round ((hi-lo)/dx);
  sum := 0;
  for i in [0..n-1] loop
    x := lo + (i+0.5)*dx;
    sum +:= dx * call(g, x);
  end loop;
  return sum;
end;
