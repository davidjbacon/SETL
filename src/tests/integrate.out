0.459699609544506
0.45969769413186
1.91541264560335e-06
